Brigthlocal/Api.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Newtonsoft.Json;
+using RestSharp;
+using System;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Security.Cryptography;
+using System.Web;
+
+namespace Brightlocal
+{
+
+    public class Api
+    {
+        /** API endpoint URL */
+        const string ENDPOINT = "https://tools.brightlocal.com/seo-tools/api";
+        /** expiry can't be more than 30 minutes (1800 seconds) */
+        const int MAX_EXPIRY = 1800;
+
+        private Uri endpoint = new Uri("https://tools.brightlocal.com/seo-tools/api/");
+        private string apiKey;
+        private string apiSecret;
+
+        // create an instance of restsharp client
+        private readonly RestClient client = new RestClient();
+
+
+        public Api(string apiKey, string apiSecret, string endpoint = null)
+        {
+            this.apiKey = apiKey;
+            this.apiSecret = apiSecret;
+            if (endpoint != null && Uri.IsWellFormedUriString(endpoint, UriKind.Absolute))
+            {
+                this.endpoint = new Uri(endpoint);
+            }
+        }
+
+        public IRestResponse Get(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.GET);
+        }
+
+        public IRestResponse Post(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.POST);
+        }
+
+        public IRestResponse Delete(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.DELETE);
+        }
+
+        public IRestResponse Put(string resource, Parameters parametrs)
+        {
+            return Call(resource, parametrs, Method.PUT);
+        }
+
+        public Batch CreateBatch(bool stopOnJobError = false, string callbackUrl = null)
+        {
+            Parameters parametrs = new Parameters();
+            parametrs.Add("stop-on-job-error", stopOnJobError);
+            if (callbackUrl != null)
+            {
+                parametrs.Add("callback", callbackUrl);
+            }
+            IRestResponse response = Post("/v4/batch", parametrs);
+            dynamic content = JsonConvert.DeserializeObject(response.Content);
```

Comment:
Why dynamic? Using your own api you should know what type of object you are expecting. I would better use  DeserializeObject<T> without dynamic.

Reply:
I think this is because the response varies by endpoint and the client is designed to be a lightweight wrapper that will work with any of our endpoints (including new ones we don't yet know about).